packages/atauth/src/atactivate.c

```diff
@@ -0,0 +1,404 @@
+#include <atcommons/enroll_status.h>
+#include <atlogger/atlogger.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+#define TAG "atactivate"
+#define DEFAULT_FIRST_APP_NAME "firstApp"
+#define DEFAULT_FIRST_DEVICE_NAME "firstDevice"
+#define AES_256_KEY_BYTES 32
+#define RSA_2048_PRIVKEY_BYTES 1300 // in PKCS#8 format includes padding
+
+int main(int argc, char *argv[]) {
+  atlogger_set_logging_level(ATLOGGER_LOGGING_LEVEL_DEBUG);
+  atlogger_set_opts(0); // disabling timestamps for now due to a potential bug in atlogger.c
+  int ret = 0;
+  char *atsign = NULL, *cram_secret = NULL, *root_host = NULL, *atkeys_fp = NULL, *otp = NULL;
+  int *root_port = NULL;
+  char enrollment_id[ENROLL_ID_MAX_LEN];
+  char status[ENROLL_STATUS_STRING_MAX_LEN];
+
+  // intialize iv used for aes encryption of keys
+  unsigned char *iv = malloc(sizeof(unsigned char) * ATCHOPS_IV_BUFFER_SIZE);
+
+  // initialize apkam symmetric key and self encryption key (bytes)
+  unsigned char *self_encryption_key_bytes, *apkam_symmetric_key_bytes;
+  size_t aes256_key_unsigned_char_bytes_size = sizeof(unsigned char) * AES_256_KEY_BYTES;
+  self_encryption_key_bytes = malloc(aes256_key_unsigned_char_bytes_size);
+  apkam_symmetric_key_bytes = malloc(aes256_key_unsigned_char_bytes_size);
+
+  // initialize base64 encoded apkam symmetric key and self encryption key
+  size_t aes_key_base64_size = atchops_base64_encoded_size(AES_256_KEY_BYTES) + 1;
+  size_t aes256_key_unsigned_char_base64_size = sizeof(unsigned char) * aes_key_base64_size;
+  unsigned char *self_encryption_key_base64 = malloc(aes256_key_unsigned_char_base64_size);
+  unsigned char *apkam_symmetric_key_base64 = malloc(aes256_key_unsigned_char_bytes_size);
+
+  // intialize encrypted APKAM symmetric Key and encrypted default encryption private key (bytes)
+  const size_t rsa_2048_privkey_base64_len = atchops_base64_encoded_size(RSA_2048_PRIVKEY_BYTES);
+  const size_t aes256_encrypted_rsa_privkey_size = atchops_aes_ctr_ciphertext_size(
+      rsa_2048_privkey_base64_len); // size for an AES256 encrypted RSA2048 privkey in bytes
+  const size_t aes256_encrypted_rsa_privkey_unsigned_char_size =
+      sizeof(unsigned char) * aes256_encrypted_rsa_privkey_size;
+  const size_t aes256_encrypted_aes256_key_size = atchops_aes_ctr_ciphertext_size(
+      aes_key_base64_size); // size of AES256 key encrypted with another AES256 key(bytes)
+  const size_t aes256_encrypted_aes256_key_unsigned_char_size =
+      sizeof(unsigned char) * aes256_encrypted_aes256_key_size;
+  unsigned char *encrypted_default_encryption_private_key = malloc(aes256_encrypted_rsa_privkey_unsigned_char_size);
+  unsigned char *encrypted_self_encryption_key = malloc(aes256_encrypted_aes256_key_unsigned_char_size);
+
+  // intialize base64 encoded encrypted APKAM symmetric Key and encrypted default encryption private key
+  const size_t aes256_encrypted_rsa_2048_privkey_base64_len = atchops_base64_encoded_size(rsa_2048_privkey_base64_len);
```

Comment:
A second aes-ctr is missing

Reply:
Can you explain why this was resolved without making the changes